src/game/RandomizationOfShipsPositions.js

```diff
@@ -0,0 +1,149 @@
+import { SHIPS_LIST } from "./constants";
+
+const GAME_BOARD = document.querySelectorAll(".board__fields");
+
+let gameBoardArray = [];
+let fieldsNamesArray = [];
+let position;
+const GAME_BOARD_WIDTH = 10;
+
+const createFieldsName = () => {
+  for (let i = 0; i < 10; i++) {
+    for (let j = 0; j < 10; j++) {
+      fieldsNamesArray.push(`${String.fromCharCode(i + 65)}${j + 1}`);
+    }
+  }
+};
+createFieldsName();
+
+const createGameBoardArray = () => {
+  gameBoardArray = [];
+  for (let i = 0; i < GAME_BOARD_WIDTH; i++) {
+    gameBoardArray.push([]);
+    for (let j = 0; j < GAME_BOARD_WIDTH; j++) {
+      gameBoardArray[i].push({
+        isEmpty: true,
+        shipName: "",
+      });
+    }
+  }
+};
+
+const rollDirection = () => {
+  return Math.floor(Math.random() * 2) === 0 ? "vertical" : "horizontal";
+};
+
+const createPosition = (shipIndex, direction) => {
+  position = [];
+  for (let i = 0; i < 2; i++) {
+    position.push(Math.floor(Math.random() * GAME_BOARD_WIDTH));
+  }
+  if (
+    direction === "horizontal" &&
+    position[1] + SHIPS_LIST[shipIndex].size > GAME_BOARD_WIDTH
+  ) {
+    position[1] = Math.abs(position[1] - SHIPS_LIST[shipIndex].size);
+  } else if (position[0] + SHIPS_LIST[shipIndex].size > GAME_BOARD_WIDTH) {
+    position[0] = Math.abs(position[0] - SHIPS_LIST[shipIndex].size);
+  }
+};
+
+const addShip = () => {
+  createGameBoardArray();
+  for (let shipIndex = 0; shipIndex < SHIPS_LIST.length; shipIndex++) {
+    const direction = rollDirection();
+    if (direction === "horizontal") {
+      positionShipHorizontally(shipIndex, direction);
+    } else {
+      positionShipVertically(shipIndex, direction);
+    }
+  }
+};
+
+const positionShipVertically = (shipIndex, direction) => {
+  createPosition(shipIndex, direction);
```

Comment:
From what I see, I was wrong earlier. You can move it to the `addShip` under `const direction = rollDirection()`. The passed `shipIndex` belong to the for from it so it shouldn't make any problem. Thanks to that you won't have in two function `createPosition` but in one. Besides, still you don't need then the `let position` at the top but creating it in the `createPosition` and return it.

Reply:
when i delete let position at the top and return it in createPosition I get a error Uncaught ReferenceError: position is not defined at positionShipVertically

Reply:
After move createPosition under const direction = rollDirection() 
RandomizationOfShipsPositions.js:76 Uncaught RangeError: Maximum call stack size exceeded
    at Array.push (<anonymous>)